server/server.js

```diff
@@ -60,10 +61,26 @@ io.on('connection', function(socket){
   socket.on('joinLobby', function(data) {
     socket.join(data.gameId);
     usersInGames[data.gameId] = usersInGames[data.gameId] || [];
-    usersInGames[data.gameId].push(data.user);
+     var playersInGame = usersInGames[data.gameId];
+     for(var i = 0; i < playersInGame.length ; i++) {
+      if(playersInGame[i].facebookId === data.user.facebookId) {
+        playersInGame.splice(i, 1);
```

Comment:
instead of splicing in the for loop, just save the index to splice after or use a flag to decide if we are going to push or not